### PR TITLE
Cleanup ToolStripItem

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup ToolStripItem

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2874)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.ToolStripItemImageIndexer.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            public override ImageList ImageList
+            {
+                get
+                {
+                    if ((_item != null) && (_item.Owner != null))
+                    {
+                        return _item.Owner.ImageList;
+                    }
+                    return null;
+                }
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.ToolStripItemImageIndexer.cs]: can be simplified down to a single line: `get => _item?.Owner?.ImageList;` (of course you can chose intermediate forms of simplification if you prefer them for stylistic reasons)

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.ToolStripItemInternalLayout.cs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                    autoSize = toolStrip.AutoSize;
+                    size = toolStrip.Size;
+                }
+                public bool IsCurrent(ToolStrip toolStrip)
+                {
+                    if (toolStrip == null)
+                    {
+                        return false;
+                    }
+                    return (toolStrip.Size == size && toolStrip.LayoutStyle == layoutStyle && toolStrip.AutoSize == autoSize);
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.ToolStripItemInternalLayout.cs]: could merge into the existing expression by prefixing it with `toolstrip != null && ...`

**Comment by RussKie** [comment]
```
##[error]System\Windows\Forms\ToolStripItem.ToolStripItemAccessibleObject.cs(181,1): error SA1028: (NETCORE_ENGINEERING_TELEMETRY=Build) Code should not contain trailing whitespace
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.cs [hunk 3]
```diff
@@ -31,149 +31,138 @@ public abstract class ToolStripItem : Component,
...
+        private static readonly int s_mergeIndexProperty = PropertyStore.CreateKey();
+
+        private static readonly int s_stateAllowDrop = BitVector32.CreateMask();
+        private static readonly int s_stateVisible = BitVector32.CreateMask(s_stateAllowDrop);
+        private static readonly int s_stateEnabled = BitVector32.CreateMask(s_stateVisible);
+        private static readonly int s_stateMouseDownAndNoDrag = BitVector32.CreateMask(s_stateEnabled);
+        private static readonly int s_stateAutoSize = BitVector32.CreateMask(s_stateMouseDownAndNoDrag);
+        private static readonly int s_statePressed = BitVector32.CreateMask(s_stateAutoSize);
+        private static readonly int s_stateSelected = BitVector32.CreateMask(s_statePressed);
+        private static readonly int stateContstructing = BitVector32.CreateMask(s_stateSelected);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.cs]:
```suggestion
        private static readonly int s_stateContstructing = BitVector32.CreateMask(s_stateSelected);
```

**Pull request merged**